store/smt/iterator.go

```diff
@@ -0,0 +1,106 @@
+package smt
+
+import (
+	"bytes"
+
+	dbm "github.com/tendermint/tm-db"
+)
+
+type Iterator struct {
+	store *Store
+	iter  dbm.Iterator
+}
+
+var (
+	indexPrefix = []byte("smt-ordering-idx-")
+	afterIndex  = []byte("smt-ordering-idx.") // '.' is next after '-' in ASCII
+)
```

Comment:
This isn't really self-explanatory. Add a brief explanation (even if it is unexported).

Nit: I'd prefer using `string` `const`ants instead and wrap them  with `[]byte()` where they are used.